packages/framework/memo-cache/README.md

```diff
@@ -0,0 +1,141 @@
+
+### getMemoCache
+
+To get an instance of the memo cache to work with, a caller starts by calling `getMemoCache`.
+
+```ts
+export function getMemoCache<T = any>(key?: object): GetMemoValue<T>;
+```
+
+This function takes an optional parameter `key` which can be an object to use as a base cache reference.
+
+- If `key` is specified, the same cache will be retrieved from the global call with the same object reference.
+- If `key` is not specified the cache instance will be unique and contained entirely within the returned function.
+
+### memoize
+
+This library also provides a standard memoization wrapper:
+
+```ts
+export function memoize<T extends Function>(fn: T): T;
+```
+
+This should be able to handle any function as an input. It will create its own instance of the cache, use any parameters to the function as key values, and return a closure with the same signature as the input.
+
+This should support the following:
+
+- Function with any number of parameters
+- Functions with no parameters
+- Any return result
+- Void functions
+
+## Examples
+
+The following are some examples for how to use the functions above for various optimizations.
+
+### Merge styles to ensure object identity does not change
+
+Given a function to merge styles together, wrap it in a memoization helper to ensure object identities don't change. Then add a helper for ensuring a CSS rule exists for a set of styles.
+
+```ts
+// standard function which will be memoized
+function mergeStylesWorker(...cssStyles: CSSStyle[]): CSSStyle {
+  // do the work of merging multiple styles together to form a new CSS style
+}
+
+// exported function internally has a caching layer with memoize
+export const mergeStyles = memoize(mergeStylesWorker);
+
+// all-in-one authoring of memoized function, this one to turn a style into a CSS class, traditionally
+// an expensive operation in browsers
+export const createRuleForStyle = memoize((style: CSSStyle) => {
+  const className = // do the work of creating the rule
+  return className;
+});
+```
+
+### Hierarchical Theme Caching
+
+This demonstrates a component called `MyComponent` that:
+
+1. has a unique cache based on the component identity
+2. cached a style computed against a theme
+3. optionally merges a style from props and caches that result
+
+These three levels of caching are effectively instance -> theme -> props.style.
+
+```ts
+import { getMemoCache } from '@fluentui-react-native/memo-cache';
+
+// get a unique cache for this component
+const myComponentCache = getMemoCache();
+
+// component is a function that takes props
+export const MyComponent = (props: IMyComponentProps) => {
+  const theme = useContext(ThemeContext);
+  const newProps = { ...props };
+
+  // get the style, cached against the theme so it will only be called once, note that because
+  const [style, nextCache] = myComponentCache(() => {
```

Comment:
to drive home the point about keys being the branching mechanism maybe call it `themeLocalCache` or something similar.

Reply:
Good idea